pkgs/build-support/rust/default.nix

```diff
@@ -121,6 +122,7 @@ in stdenv.mkDerivation (args // {
   checkPhase = args.checkPhase or ''
     runHook preCheck
     echo "Running cargo test"
+    export CARGO_HOME=$(pwd)
```

Comment:
I would rather set it to `$TMPDIR` though. And it only needs to be set once - maybe in the `configurePhase`?